### DISCUSSION TITLE
Remote cache

### DISCUSSION THREAD

**Discussion by povder** [ScalablyTyped/Converter#393] [General]
Hi, I know that ScalablyTyped already supports remote cache with its `stRemoteCache` setting, but I was thinking, now that sbt [supports](https://www.scala-sbt.org/1.x/docs/Remote-Caching.html) remote cache for its incremental compilation could that sbt's mechanism be leveraged somehow to do remote caching in ScalablyTyped? I'm asking because that would open up the possibility of leveraging pre-existing resolver implementations and so ScalablyTyped 's artifact could be pushed to a wide range of destinations, not just ones implemented for ScalablyTyped.

What are the maintainer's thoughts on this?

**Comment by oyvindberg** [comment]
I think that sounds great! 

Do you have any idea if we're free to upload any kind of files to the sbt remote cache? in the case of ST it's a bunch of json files along with jars.

I should also add that I won't have time to work on this myself, but I'd certainly be happy to help out if you want to contribute :)

**Comment by povder** [comment]
> Do you have any idea if we're free to upload any kind of files to the sbt remote cache?

Not sure if one can call that a hack but sbt actually packages everything it needs into a JAR file, likely so that the entire "resolver machinery" can be re-used. It probably needs to build something that resembles an Ivy artifact. That "JAR" contains compiled classes but also a bunch of other things.

> I should also add that I won't have time to work on this myself, but I'd certainly be happy to help out if you want to contribute :)

Yeah, I'm considering contributing :)

**Reply by oyvindberg** [reply]
Right. shame if it has to be one huge jar for all the files, because right now everything is nicely deduplicated with checksums and whatnot. but definitely do investigate if you have time, it would be much better to lean on sbt for this

**Comment by povder** [comment]
I should clarify - sbt does a single JAR per sbt project so I'm hoping in case of scalablytyped it would be a single JAR per NPM module.

**Comment by oyvindberg** [comment]
Alright that would be perfect :)
